pkg/prometheus/agent/operator.go

```diff
@@ -550,13 +551,24 @@ func (c *Operator) addHandlers() {
 }
 
 // Resolve implements the operator.Syncer interface.
-func (c *Operator) Resolve(ss *appsv1.StatefulSet) metav1.Object {
-	key, ok := c.accessor.MetaNamespaceKey(ss)
+func (c *Operator) Resolve(obj interface{}) metav1.Object {
+	key, ok := c.accessor.MetaNamespaceKey(obj)
 	if !ok {
 		return nil
 	}
 
-	match, promKey := statefulSetKeyToPrometheusAgentKey(key)
+	var match bool
+	var promKey string
+
+	switch obj.(type) {
+	case *appsv1.DaemonSet:
+		if c.daemonSetFeatureGateEnabled {
+			match, promKey = daemonSetKeyToPrometheusAgentKey(key)
+		}
```

Comment:
I still need to investigate this, but isn't the daemonset informer only created if the featuregate is enabled? I.e. it's impossible to `Resolve` to be called by a daemonset change if the featuregate is disabled.

If what I'm saying is correct, we probably can remove the conditional here 🤔

Reply:
I've just tested locally without the conditional and it seems to have worked without problems 🤔

Reply:
Yeah makes sense. I'll remove this condition